src/main/java/org/wso2/carbon/connector/iso8583/ISO8583MessageHandler.java

```diff
@@ -56,28 +57,31 @@ public ISO8583MessageHandler(MessageContext messageContext, String details, Stri
     /**
      * handle the iso8583 message request and responses
      *
-     * @param isoMessage  packed ISOMessage
-     * @param connection  Socket connection with backend Test server
      * @param messageContext the message context
+     * @param connection  Socket connection with backend Test server
+     * @param isoMessage  packed ISOMessage
      */
-    public void clientHandler(MessageContext messageContext, Socket connection, String isoMessage) {
+    public void clientHandler(MessageContext messageContext, Socket connection, byte[] isoMessage) {
         DataOutputStream outStream = null;
-        BufferedReader inFromServer = null;
-        String message;
+        DataInputStream inFromServer = null;
         try {
             outStream = new DataOutputStream(connection.getOutputStream());
-            inFromServer = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            inFromServer = new DataInputStream(connection.getInputStream());
             if (connection.isConnected()) {
-                outStream.writeUTF(isoMessage);
+                outStream.write(isoMessage);
                 outStream.flush();
 
                 /* Sender will receive the Acknowledgement here */
-                if ((message = inFromServer.readLine()) != null) {
-                    unpackResponse(messageContext, message);
-                }
+                Thread.sleep(200);
```

Comment:
Why did you put 200 sleep here?

Reply:
It will take time to get the response from the back end in the input stream buffer. That's why I put.

Reply:
Please check without it ...